Convert CUSP to signed distance

mark-petersen opened the pull request:
Re-design of CUSP mesh using a number of overlaid regions. Each region begins with a signed distance from a geojson region file. A mask is computed using a tanh function on the signed distance.

mark-petersen commented:
@xylar, it would be great to hear your thoughts on this. The details are preliminary, but the process is in place. It automatically generates this plot, which shows how there are five masks, and each one adds to the high resolution region. There are very few points in each geojson file (also added to this PR) so the regions are easy to alter.
![mesh_construction](https://user-images.githubusercontent.com/4036816/79628090-268ad200-80fb-11ea-9701-f3266485ea70.png)
![cellWidthGlobal](https://user-images.githubusercontent.com/4036816/79628093-2be81c80-80fb-11ea-9398-62fe03e788de.png)

xylar commented on testing_and_setup/compass/ocean/jigsaw_to_MPAS/mesh_definition_tools.py:
```diff
@@ -65,7 +65,16 @@ def mergeCellWidthVsLat(
 
 
-def EC_CellWidthVsLat(lat):
+def EC_CellWidthVsLat(lat,
+    cellWidthEq = 30.0,
+    cellWidthMidLat = 60.0,
+    cellWidthPole = 35.0,
+    latPosEq = 15.0,
+    latPosPole = 73.0,
+    latTransition = 40,
+    latWidthEq = 6.0,
+    latWidthPole = 9.0
+    ):
```
@mark-petersen, I did this work in #518 so we're competing with one another on this.  I made the docstring have numpydoc formatting (allowing it to be automatically parsed and formatted the arguments in a more standard way -- one line per argument really isn't a standard practice.  In any case, we need to reconcile these differences before the next mega-merge.

mark-petersen replied:
done.

xylar commented on testing_and_setup/compass/ocean/global_ocean/CUSP12/init/define_base_mesh.py:
```diff
@@ -1,187 +1,375 @@
+    # Create cell width vs latitude for Atlantic and Pacific basins
+    QU1 = np.ones(lat.size)
+    EC60to30 = mdt.EC_CellWidthVsLat(lat)*km
+    EC60to30Narrow = mdt.EC_CellWidthVsLat(lat, latPosEq = 8.0, latWidthEq = 3.0)*km
```
@vanroekel seems to think in #522 that we want the equator region to be at least 10 degrees wide (probably with a 4.0 degree transition for consistency).  I think we want to be consistent between all the meshes, so this should probably do the same, as should #518

xylar commented on testing_and_setup/compass/ocean/global_ocean/CUSP12/init/define_base_mesh.py:
```diff
@@ -1,187 +1,375 @@
+    fileName = 'coastline_CUSP'
+    transitionOffset = 800.0*km
+    transitionWidth = 600.0*km
+    fc = read_feature_collection(fileName + '.geojson')
```
```suggestion
    fc = read_feature_collection('{}.geojson'.format(fileName))
```

xylar replied:
And similarly elsewhere.

mark-petersen replied:
done, thanks.

xylar commented:
Please rebase onto #526 and #527 (and make any changes to those PRs if you're not happy with them).  Otherwise, we're competing with one another.

mark-petersen commented:
Rebased on ocean/develop. This takes care of @xylar's requests above.

mark-petersen commented:
Rebased and updated.
![cellWidthGlobal](https://user-images.githubusercontent.com/4036816/80603800-1e317180-89ee-11ea-866d-126278a72cb8.png)
![mesh_construction](https://user-images.githubusercontent.com/4036816/80603806-1ffb3500-89ee-11ea-9327-f17f1895cade.png)

mark-petersen commented:
@xylar, I would like to have two PRs for the CUSP mesh. This one is just implementing the signed distance function and the plots, and it is complete. Then I will open a new one with the mesh label, that can be open for discussion for a while. That way we separate the first implementation from the group discussions.

You already reviewed it once, but I'm happy to hear further comments. Otherwise, please approve and I'll merge it in.
